[400]Feature/notifications and mail triggers

hemant9956 opened the pull request:
What: 

Why:

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

hemant9956 committed:
Remove storj binary file and integrate mail service for user account notifications

- Deleted the `storj` binary file as part of the cleanup process.
- Enhanced the admin server to include a `mailService` for sending email notifications upon user account activation and deactivation.
- Implemented email templates for account activation and deactivation notifications, improving user communication.
- Updated the user account management functions to send email notifications asynchronously, ensuring a better user experience.

hemant9956 committed:
Enhance accounting and notification services

- Added a new `ProjectUsage` service to the `Core` struct for improved accounting functionality.
- Refactored email reminders setup to utilize the new `ProjectUsage` service, ensuring accurate bandwidth usage notifications.
- Introduced new email templates for user registration and plan purchase confirmations, enhancing user communication.
- Updated the `Chore` struct to include bandwidth usage reminders, sending push notifications when usage exceeds thresholds.
- Implemented push notifications for various events, including API key creation and deletion, data sharing, and access grants, improving user engagement.

princeparmar commented on satellite/admin/user.go:
```diff
@@ -2483,6 +2494,56 @@ func (server *Server) deactivateUserAccount(w http.ResponseWriter, r *http.Reque
 
+	// Send email notification for account deactivated
+	if server.mailService != nil {
+		go func() {
```
move that to different function

princeparmar commented on satellite/admin/user.go:
```diff
@@ -2553,6 +2614,74 @@ func (server *Server) activateUserAccount(w http.ResponseWriter, r *http.Request
 	}
 
+	// Send push notification for account activated
+	go func() {
```
fix structure

princeparmar commented on satellite/admin/user.go:
```diff
@@ -2553,6 +2614,74 @@ func (server *Server) activateUserAccount(w http.ResponseWriter, r *http.Request
+
+	// Send email notification for account activated
+	if server.mailService != nil {
+		go func() {
```
+

hemant9956 committed:
Refactor push notification handling and improve user engagement

- Replaced direct push notification calls with a new `SendNotificationAsync` method for asynchronous processing, enhancing responsiveness.
- Consolidated notification logic into a single method, `sendPushNotificationByEventName`, to streamline the sending process and reduce redundancy.
- Updated various services to utilize the new notification methods, ensuring consistent handling of user notifications across the application.
- Enhanced logging for notification success and failure cases, improving monitoring and debugging capabilities.
- Implemented user preference checks for notifications, allowing for more tailored user experiences based on individual settings.

hemant9956 committed:
Refactor push notification logic and remove unused methods

- Simplified the `SendPushNotificationWithPreferences` method by consolidating the active push configuration retrieval into a separate function, improving code clarity.
- Removed the `shouldSendNotification` method, streamlining the notification handling process and reducing redundancy in the codebase.
- Cleaned up the `extractIPFromRequest` function, which was previously unused, enhancing overall code maintainability.
